### PR TITLE
Add a document method that returns a layer's bounding box

### LINKED ISSUE #151: Unify bounding box for shapes during creation
Provide a mechanism to calculate and update the bounding box of a shape while drawing it.
This will honour things like the Center or LockAspectRatio modifiers.

If done before #148, there will be more to refactor in that issue. However this is higher priority.

Complexity: 2
Involves: Editor (Rust)

### PR DESCRIPTION
**Pull request by AyeTbk**
Closes #151

---
This change is [Reviewable](https://reviewable.io/reviews/graphiteeditor/graphite/273)

### CODE REVIEW

File: core/document/src/layers/mod.rs [hunk 1]
```diff
@@ -100,6 +100,13 @@ impl LayerDataTypes {
 			}
 		}
 	}
+
+	pub fn bounding_box(&self, transform: glam::DAffine2, style: style::PathStyle) -> [DVec2; 2] {
+		use kurbo::Shape;
```
Comment (henryksloan) [inline core/document/src/layers/mod.rs:105]: Move `use` to the top of the file.

**Review by henryksloan — APPROVED** [review]
LGTM

File: core/document/src/layers/mod.rs (hunk 1 above)
Comment (AyeTbk) [reply core/document/src/layers/mod.rs]: done

File: core/document/src/document.rs [hunk 2]
```diff
@@ -227,6 +227,11 @@ impl Document {
 		Ok(())
 	}
 
+	pub fn layer_bounding_box(&self, path: &[LayerId]) -> Result<[DVec2; 2], DocumentError> {
```
Comment (Keavon) [inline core/document/src/document.rs:230]: Please rename `layer_axis_aligned_bounding_box` and create another function called `layer_local_bounding_box` which returns the bounding box of the layer's own rotation.

Comment (AyeTbk) [reply core/document/src/document.rs]: Done

**Review by Keavon — APPROVED** [review]

**Pull request merged**